aapenimservice/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/aapenimservice/LagreSelvbestemtImService.kt

```diff
@@ -75,46 +77,42 @@ class LagreSelvbestemtImService(
 
     override fun new(melding: Map<Key, JsonElement>) {
         val transaksjonId = Key.UUID.les(UuidSerializer, melding)
-        val selvbestemtId = Key.SELVBESTEMT_ID.les(UuidSerializer, melding)
-        val skjema = Key.SKJEMA_INNTEKTSMELDING.les(SkjemaInntektsmelding.serializer(), melding)
-        val avsenderFnr = Key.ARBEIDSGIVER_FNR.les(String.serializer(), melding)
+        val skjema = Key.SKJEMA_INNTEKTSMELDING.les(SkjemaInntektsmeldingSelvbestemt.serializer(), melding)
+        val avsenderFnr = Key.ARBEIDSGIVER_FNR.les(Fnr.serializer(), melding)
 
         MdcUtils.withLogFields(
             Log.klasse(this),
             Log.event(event),
-            Log.transaksjonId(transaksjonId),
-            Log.selvbestemtId(selvbestemtId)
+            Log.transaksjonId(transaksjonId)
         ) {
-            rapid.publish(
+            rapid.publishNotNull(
                 Key.EVENT_NAME to event.toJson(),
                 Key.UUID to transaksjonId.toJson(),
-                Key.SELVBESTEMT_ID to selvbestemtId.toJson(),
+                Key.SELVBESTEMT_ID to skjema.selvbestemtId?.toJson(),
                 Key.BEHOV to BehovType.VIRKSOMHET.toJson(),
-                Key.ORGNRUNDERENHET to skjema.avsender.orgnr.toJson()
+                Key.ORGNRUNDERENHET to skjema.avsender.orgnr.toJson(Orgnr.serializer())
             )
 
-            rapid.publish(
+            rapid.publishNotNull(
                 Key.EVENT_NAME to event.toJson(),
                 Key.UUID to transaksjonId.toJson(),
-                Key.SELVBESTEMT_ID to selvbestemtId.toJson(),
+                Key.SELVBESTEMT_ID to skjema.selvbestemtId?.toJson(),
                 Key.BEHOV to BehovType.HENT_PERSONER.toJson(),
                 Key.FNR_LISTE to listOf(
                     skjema.sykmeldtFnr,
                     avsenderFnr
-                ).toJson(String.serializer())
+                ).toJson(Fnr.serializer())
             )
         }
     }
 
     override fun inProgress(melding: Map<Key, JsonElement>) {
         val transaksjonId = Key.UUID.les(UuidSerializer, melding)
-        val selvbestemtId = Key.SELVBESTEMT_ID.les(UuidSerializer, melding)
 
         MdcUtils.withLogFields(
             Log.klasse(this),
             Log.event(event),
-            Log.transaksjonId(transaksjonId),
-            Log.selvbestemtId(selvbestemtId)
```

Comment:
betyr dette at man må gå i tjenestekall for å se selvbestemt id i loggen?

Reply:
Nei, det betyr bare at selvbestemt-ID ikke vil komme som et felt i loggene, både i åpne og sikre logger. Jeg tenker det kan legges tilbake på sikt, men da må jeg flikke litt på loggfeltkoden også, så tar det senere.